app/src/lib/git/checkout.ts

```diff
@@ -66,44 +115,60 @@ export async function checkoutBranch(
   branch: Branch,
   progressCallback?: ProgressCallback
 ): Promise<true> {
-  let opts: IGitExecutionOptions = {
-    env: await envForRemoteOperation(
-      account,
-      getFallbackUrlForProxyResolve(account, repository)
-    ),
-    expectedErrors: AuthenticationErrors,
-  }
+  const opts = await getCheckoutOpts(
+    repository,
+    account,
+    `Checking out branch ${branch.name}`,
+    branch.name,
+    `Switching to ${__DARWIN__ ? 'Branch' : 'branch'}`,
+    progressCallback
+  )
 
-  if (progressCallback) {
-    const title = `Checking out branch ${branch.name}`
-    const kind = 'checkout'
-    const targetBranch = branch.name
-
-    opts = await executionOptionsWithProgress(
-      { ...opts, trackLFSProgress: true },
-      new CheckoutProgressParser(),
-      progress => {
-        if (progress.kind === 'progress') {
-          const description = progress.details.text
-          const value = progress.percent
-
-          progressCallback({ kind, title, description, value, targetBranch })
-        }
-      }
-    )
+  const baseArgs = getCheckoutArgs(progressCallback)
+  const args = [...baseArgs, ...(await getBranchCheckoutArgs(branch))]
 
-    // Initial progress
-    progressCallback({ kind, title, value: 0, targetBranch })
-  }
+  await git(args, repository.path, 'checkoutBranch', opts)
 
-  const args = await getCheckoutArgs(
+  // we return `true` here so `GitStore.performFailableGitOperation`
+  // will return _something_ differentiable from `undefined` if this succeeds
+  return true
+}
+
+/**
+ * Check out the given commit.
+ * Literally invokes `git checkout <commit SHA>`.
+ *
+ * @param repository - The repository in which the branch checkout should
+ *                     take place
+ *
+ * @param commit     - The commit that should be checked out
+ *
+ * @param progressCallback - An optional function which will be invoked
+ *                           with information about the current progress
+ *                           of the checkout operation. When provided this
+ *                           enables the '--progress' command line flag for
+ *                           'git checkout'.
+ */
+export async function checkoutCommit(
+  repository: Repository,
+  account: IGitAccount | null,
+  commit: CommitOneLine,
+  progressCallback?: ProgressCallback
+): Promise<true> {
+  const title = `Checking out ${__DARWIN__ ? 'Commit' : 'commit'}`
+  const opts = await getCheckoutOpts(
     repository,
-    branch,
     account,
+    title,
+    shortenSHA(commit.sha),
+    title,
```

Comment:
Looking at the implementation, I guess this could be `undefined`? That's probably less confusing than seeing `title` twice (it definitely made me wonder why 😳 )
```suggestion
    undefined,
```